cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -345,7 +348,7 @@ func (cm *certManagerClient) shouldUpgrade(objs []unstructured.Unstructured) (st
 			break
 		}
 	}
-	return currentVersion, config.Version(), needUpgrade, nil
+	return currentVersion, configVersion, needUpgrade, nil
```

Comment:
Could we use only semver.Version everywhere and avoid carrying strings as well?

Reply:
`semver.Version` doesn't have a field with the unparsed semver
I could use the `String()` method to regenerate it whenever it's needed, but that seems a bit unnecessary, considering we already have it